### ISSUE TITLE
Falling throught the world

### ISSUE THREAD

**Issue by MrEDok** [Imprex-Development/orebfuscator#237]
### Description

I don't know if it problem with replay mod or Orebfuscator but when you relog while riding a horse (only mod that we tested) you will fall 100 orso block below where you relog while using replay mod.

Server: git-Purpur-1768 (MC: 1.19.2)*
Plugin: Orebfuscator version 5.3.0

### Reproduction Steps

1.19.2 Client

Client: have replay mod on.

Server:
config: 
**Enable proximity**
> obfuscation:
  obfuscation-overworld:
    enabled: false
    proximity:
  proximity-overworld:
    enabled: true
    `
    

1. Now join the server with replay mod
2. Tame a horse
3. Ride the horse
4. Disconnect from the server while riding a horse
5. log back on and you will be "teleported" below bedrock or wayway off on Y level where you relog
    
https://streamable.com/4zqm1w

### Expected Behaviour

Not teleporting the player down.

### Server Details

git-Purpur-1768 (MC: 1.19.2)*

### Orebfuscator Version

5.3.0

### ProtocolLib Version

5.0.0-SNAPSHOT-b600

### Logs and Configs
<details>
  <summary>Config</summary>
```yml

version: 2
general:
  checkForUpdates: true
  updateOnBlockDamage: true
  bypassNotification: true
  ignoreSpectator: false
  updateRadius: 2
advanced:
  verbose: false
  useAsyncPacketListener: true
  maxMillisecondsPerTick: 10
  protocolLibThreads: -1
  obfuscationWorkerThreads: -1
  proximityHiderThreads: -1
cache:
  enabled: true
  baseDirectory: orebfuscator_cache
  maximumOpenRegionFiles: 256
  deleteRegionFilesAfterAccess: 172800000
  maximumSize: 16384
  expireAfterAccess: 60000
  maximumTaskQueueSize: 32768
obfuscation:
  obfuscation-overworld:
    enabled: true
    minY: -2032
    maxY: 2031
    worlds:
    - world
    hiddenBlocks:
    - minecraft:emerald_ore
    - minecraft:deepslate_emerald_ore
    - minecraft:diamond_ore
    - minecraft:deepslate_diamond_ore
    - minecraft:gold_ore
    - minecraft:deepslate_gold_ore
    - minecraft:raw_gold_block
    - minecraft:iron_ore
    - minecraft:deepslate_iron_ore
    - minecraft:raw_iron_block
    - minecraft:coal_ore
    - minecraft:deepslate_coal_ore
    - minecraft:lapis_ore
    - minecraft:deepslate_lapis_ore
    - minecraft:redstone_ore
    - minecraft:deepslate_redstone_ore
    - minecraft:copper_ore
    - minecraft:raw_copper_block
    - minecraft:deepslate_copper_ore
    - minecraft:stone
    - minecraft:deepslate
    - minecraft:chest
    - minecraft:ender_chest
    - minecraft:trapped_chest
    - minecraft:shulker_box
    - minecraft:white_shulker_box
    - minecraft:orange_shulker_box
    - minecraft:magenta_shulker_box
    - minecraft:light_blue_shulker_box
    - minecraft:yellow_shulker_box
    - minecraft:lime_shulker_box
    - minecraft:pink_shulker_box
    - minecraft:gray_shulker_box
    - minecraft:light_gray_shulker_box
    - minecraft:cyan_shulker_box
    - minecraft:purple_shulker_box
    - minecraft:blue_shulker_box
    - minecraft:brown_shulker_box
    - minecraft:green_shulker_box
    - minecraft:red_shulker_box
    - minecraft:black_shulker_box
    randomBlocks:
      minecraft:stone: 30
      minecraft:deepslate: 30
      minecraft:cave_air: 5
      minecraft:cobblestone: 1
      minecraft:diamond_ore: 25
      minecraft:emerald_ore: 10
      minecraft:gold_ore: 5
      minecraft:iron_ore: 5
      minecraft:copper_ore: 5
      minecraft:coal_ore: 5
      minecraft:lapis_ore: 5
      minecraft:redstone_ore: 5
      minecraft:mossy_cobblestone: 1
      minecraft:obsidian: 1
      minecraft:tnt: 1
  obfuscation-nether:
    enabled: true
    minY: -2032
    maxY: 2031
    worlds:
    - world_nether
    hiddenBlocks:
    - minecraft:netherrack
    - minecraft:nether_quartz_ore
    - minecraft:chest
    - minecraft:ender_chest
    - minecraft:trapped_chest
    - minecraft:shulker_box
    - minecraft:white_shulker_box
    - minecraft:orange_shulker_box
    - minecraft:magenta_shulker_box
    - minecraft:light_blue_shulker_box
    - minecraft:yellow_shulker_box
    - minecraft:lime_shulker_box
    - minecraft:pink_shulker_box
    - minecraft:gray_shulker_box
    - minecraft:light_gray_shulker_box
    - minecraft:cyan_shulker_box
    - minecraft:purple_shulker_box
    - minecraft:blue_shulker_box
    - minecraft:brown_shulker_box
    - minecraft:green_shulker_box
    - minecraft:red_shulker_box
    - minecraft:black_shulker_box
    - minecraft:ancient_debris
    - minecraft:nether_gold_ore
    - minecraft:basalt
    - minecraft:soul_sand
    - minecraft:soul_soil
    randomBlocks:
      minecraft:netherrack: 4
      minecraft:nether_quartz_ore: 1
      minecraft:nether_gold_ore: 1
      minecraft:ancient_debris: 1
  obfuscation-end:
    enabled: true
    minY: -2032
    maxY: 2031
    worlds:
    - world_the_end
    hiddenBlocks:
    - minecraft:chest
    - minecraft:ender_chest
    - minecraft:trapped_chest
    - minecraft:shulker_box
    - minecraft:white_shulker_box
    - minecraft:orange_shulker_box
    - minecraft:magenta_shulker_box
    - minecraft:light_blue_shulker_box
    - minecraft:yellow_shulker_box
    - minecraft:lime_shulker_box
    - minecraft:pink_shulker_box
    - minecraft:gray_shulker_box
    - minecraft:light_gray_shulker_box
    - minecraft:cyan_shulker_box
    - minecraft:purple_shulker_box
    - minecraft:blue_shulker_box
    - minecraft:brown_shulker_box
    - minecraft:green_shulker_box
    - minecraft:red_shulker_box
    - minecraft:black_shulker_box
    randomBlocks:
      minecraft:end_stone: 1
      minecraft:end_stone_bricks: 1
      minecraft:bedrock: 1
      minecraft:obsidian: 1
      minecraft:purpur_block: 1
proximity:
  proximity-overworld:
    enabled: true
    minY: -2032
    maxY: 2031
    worlds:
    - world
    distance: 24
    useFastGazeCheck: false
    useBlockBelow: true
    hiddenBlocks:
      minecraft:mossy_cobblestone: {}
      minecraft:diamond_ore: {}
      minecraft:deepslate_diamond_ore: {}
      minecraft:emerald_ore: {}
      minecraft:deepslate_emerald_ore: {}
      minecraft:chest: {}
      minecraft:ender_chest: {}
      minecraft:trapped_chest: {}
      minecraft:anvil: {}
      minecraft:crafting_table: {}
      minecraft:dispenser: {}
      minecraft:enchanting_table: {}
      minecraft:furnace: {}
      minecraft:blast_furnace: {}
      minecraft:cartography_table: {}
      minecraft:fletching_table: {}
      minecraft:grindstone: {}
      minecraft:composter: {}
      minecraft:lectern: {}
      minecraft:loom: {}
      minecraft:smithing_table: {}
      minecraft:smoker: {}
      minecraft:stonecutter: {}
      minecraft:hopper: {}
      minecraft:spawner: {}
      minecraft:shulker_box: {}
      minecraft:white_shulker_box: {}
      minecraft:orange_shulker_box: {}
      minecraft:magenta_shulker_box: {}
      minecraft:light_blue_shulker_box: {}
      minecraft:yellow_shulker_box: {}
      minecraft:lime_shulker_box: {}
      minecraft:pink_shulker_box: {}
      minecraft:gray_shulker_box: {}
      minecraft:light_gray_shulker_box: {}
      minecraft:cyan_shulker_box: {}
      minecraft:purple_shulker_box: {}
      minecraft:blue_shulker_box: {}
      minecraft:brown_shulker_box: {}
      minecraft:green_shulker_box: {}
      minecraft:red_shulker_box: {}
      minecraft:black_shulker_box: {}
      minecraft:bee_nest: {}
      minecraft:beehive: {}
    randomBlocks:
      minecraft:stone: 1
  proximity-nether:
    enabled: true
    minY: -2032
    maxY: 2031
    worlds:
    - world_nether
    distance: 24
    useFastGazeCheck: false
    useBlockBelow: true
    hiddenBlocks:
      minecraft:chest: {}
      minecraft:ender_chest: {}
      minecraft:trapped_chest: {}
      minecraft:anvil: {}
      minecraft:crafting_table: {}
      minecraft:dispenser: {}
      minecraft:enchanting_table: {}
      minecraft:furnace: {}
      minecraft:blast_furnace: {}
      minecraft:cartography_table: {}
      minecraft:fletching_table: {}
      minecraft:grindstone: {}
      minecraft:composter: {}
      minecraft:lectern: {}
      minecraft:loom: {}
      minecraft:smithing_table: {}
      minecraft:smoker: {}
      minecraft:stonecutter: {}
      minecraft:hopper: {}
      minecraft:spawner: {}
      minecraft:shulker_box: {}
      minecraft:white_shulker_box: {}
      minecraft:orange_shulker_box: {}
      minecraft:magenta_shulker_box: {}
      minecraft:light_blue_shulker_box: {}
      minecraft:yellow_shulker_box: {}
      minecraft:lime_shulker_box: {}
      minecraft:pink_shulker_box: {}
      minecraft:gray_shulker_box: {}
      minecraft:light_gray_shulker_box: {}
      minecraft:cyan_shulker_box: {}
      minecraft:purple_shulker_box: {}
      minecraft:blue_shulker_box: {}
      minecraft:brown_shulker_box: {}
      minecraft:green_shulker_box: {}
      minecraft:red_shulker_box: {}
      minecraft:black_shulker_box: {}
      minecraft:bee_nest: {}
      minecraft:beehive: {}
      minecraft:ancient_debris: {}
      minecraft:nether_gold_ore: {}
      minecraft:respawn_anchor: {}
    randomBlocks:
      minecraft:netherrack: 1
  proximity-end:
    enabled: true
    minY: -2032
    maxY: 2031
    worlds:
    - world_the_end
    distance: 24
    useFastGazeCheck: false
    useBlockBelow: true
    hiddenBlocks:
      minecraft:chest: {}
      minecraft:ender_chest: {}
      minecraft:trapped_chest: {}
      minecraft:anvil: {}
      minecraft:crafting_table: {}
      minecraft:dispenser: {}
      minecraft:enchanting_table: {}
      minecraft:furnace: {}
      minecraft:blast_furnace: {}
      minecraft:cartography_table: {}
      minecraft:fletching_table: {}
      minecraft:grindstone: {}
      minecraft:composter: {}
      minecraft:lectern: {}
      minecraft:loom: {}
      minecraft:smithing_table: {}
      minecraft:smoker: {}
      minecraft:stonecutter: {}
      minecraft:hopper: {}
      minecraft:spawner: {}
      minecraft:shulker_box: {}
      minecraft:white_shulker_box: {}
      minecraft:orange_shulker_box: {}
      minecraft:magenta_shulker_box: {}
      minecraft:light_blue_shulker_box: {}
      minecraft:yellow_shulker_box: {}
      minecraft:lime_shulker_box: {}
      minecraft:pink_shulker_box: {}
      minecraft:gray_shulker_box: {}
      minecraft:light_gray_shulker_box: {}
      minecraft:cyan_shulker_box: {}
      minecraft:purple_shulker_box: {}
      minecraft:blue_shulker_box: {}
      minecraft:brown_shulker_box: {}
      minecraft:green_shulker_box: {}
      minecraft:red_shulker_box: {}
      minecraft:black_shulker_box: {}
      minecraft:bee_nest: {}
      minecraft:beehive: {}
    randomBlocks:
      minecraft:end_stone: 1
```
</details>

### Extra Details

[_No response_](https://streamable.com/4zqm1w)

**Comment by NgLoader** [comment]
When you disable OFC, is it still bugged?
You can try to disable the async cache.

**Comment by MrEDok** [comment]
> When you disable OFC, is it still bugged? You can try to disable the async cache.

When disable Orebfuscator player doesn't fall throught the world anymore.

Assuming to disable async cache is this
cache:
  enabled: false
  Result: Still falling throught the world

useAsyncPacketListener: false
  Result: Still falling throught the world
  
  
  Abit more testing:
  This seem to only happen with HORSE and SKELETON_HORSE **WITH** SADDLE if without saddle you wont fall throught the world

**Comment by NgLoader** [comment]
I check it tomorrow

**Comment by NgLoader** [comment]
Please change your server version to spigot or paper.
Or report your issue to Purpur, it is not a OFC error.

**Comment by Ingrim4** [comment]
Orebfuscator only modifies the chunk packets without change anything server-side which means the server should be able to stop the player from falling through the world. This can only mean that Purpur doesn't check the players position and allows the player to go through walls or in your case the floor. Nothing we can do about it since we have to modify the packets async in order to not lag the server too much. You could try Spigot/Paper and see if one of those works better.

**Comment by MrEDok** [comment]
> 

This also happen on paper `git-Paper-135 (MC: 1.19.2)`
This does not happen on spigot `CraftBukkit version 3579-Spigot-e53686f-dd0a218 (MC: 1.19.2)`

So i assume it something todo with paper chunk system.

**Comment by Ingrim4** [comment]
Should be fixed as part of dmulloy2/ProtocolLib#2545
